php/class-post-type.php

```diff
@@ -312,4 +351,37 @@ public function add_editor_color_palette() {
 			]
 		);
 	}
+
+	/**
+	 * Admin page post meta query.
+	 *
+	 * @param array  $args    filter args.
+	 * @param object $request filter request.
+	 * @return $args
+	 */
+	public function admin_page_sub_menu_query( $args, $request ) {
+		$meta_key = $request->get_param( 'metaKey' );
+		if ( $meta_key ) {
+			$args['meta_key']   = $meta_key;
+			$args['meta_value'] = $request->get_param( 'metaValue' );
+		}
+		return $args;
+	}
+
+	/**
+	 * Add default value to sub menu meta.
+	 *
+	 * @param integer $post_id  filter args.
+	 * @param object  $post     filter request.
+	 */
+	public function add_default_sub_menu_meta_to_post( $post_id, $post ) {
+		if ( $post->post_type !== $this->slug ) {
+			return;
+		}
+
+		$value = get_post_meta( $post_id, $this->sub_menu_key, true );
+		if ( 0 === $value ) {
+			update_post_meta( $post_id, $this->sub_menu_key, 0 );
+		}
```

Comment:
I don't understand what this block is trying to do. It looks like:

if 0 is the saved meta value, save zero as the meta value.

Reply:
I agree with you, this looks very strange. In order for the query to work in meta-select-control.js to work, every parent admin page needed a postmeta key of sub_menu with a value of 0. However, if you just created a page, and didn't select it to be a sub_menu, it would not save any value for sub_menu and therefore be excluded from the query. To fix this, I added this method to check every admin-page that is saved and assign it a zero value if it was not there.
`const query = {
			per_page: -1,
			exclude: currentPostId,
			metaKey: 'sub_menu',
			metaValue: 0,
		};`
Now the reason it looks strange is because get_post_meta( $post_id, $this->sub_menu_key, true ); returns a value of 0 if the post meta key does not exist at all. However, it will obviously also return zero if there is a value of zero already.

Does that make sense? Not sure how else to handle it. I tried adding default => 0 when registering the sub_menu meta but found that it also doesn't automatically add it to the post.

If we get the global $menu query working above then I can do away with this whole thing.

Reply:
Can it be done away with now?

Reply:
Yip. Gone with the wind now